gateway/src/cluster/impl.rs

```diff
@@ -142,26 +142,25 @@ impl Cluster {
             metrics::gauge!("Cluster-Shard-Count", total.try_into().unwrap_or(-1));
         }
 
-        let mut shards = HashMap::new();
+        let shards = (from..=to)
+            .map(|idx| {
+                let mut shard_config = config.shard_config().clone();
+                shard_config.shard = [idx, total];
 
-        for idx in from..=to {
-            let mut shard_config = config.shard_config().clone();
-            shard_config.shard = [idx, total];
-            let resume_sessions = config.resume_sessions().get(&idx);
+                if let Some(data) = config.resume_sessions().get(&idx) {
+                    shard_config.session_id = Some(data.session_id.clone());
```

Comment:
Would it be possible to just call `resume_sessions().remove(&idx)` here instead? It doesn't look like the resume data is needed anywhere after this loop.